examples/tic-tac-toe/ui/src/hooks/useTransactions.ts

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { bcs } from '@mysten/sui/bcs';
+import { PublicKey } from '@mysten/sui/cryptography';
+import { ObjectRef, Transaction } from '@mysten/sui/transactions';
+import { useNetworkVariable } from 'config';
+import { Game } from 'hooks/useGameQuery';
+import { TurnCap } from 'hooks/useTurnCapQuery';
+import { multiSigPublicKey } from 'MultiSig';
+
+/** Hook to provide an instance of the Transactions builder. */
+export function useTransactions(): Transactions | null {
+	const packageId = useNetworkVariable('packageId');
+	return packageId ? new Transactions(packageId) : null;
+}
+
+/**
+ * Builds on-chain transactions for the Tic-Tac-Toe game.
+ */
+export class Transactions {
+	readonly packageId: string;
+
+	constructor(packageId: string) {
+		this.packageId = packageId;
+	}
+
+	newSharedGame(player: string, opponent: string): Transaction {
+		const tx = new Transaction();
+
+		tx.moveCall({
+			target: `${this.packageId}::shared::new`,
+			arguments: [tx.pure.address(player), tx.pure.address(opponent)],
+		});
+
+		return tx;
+	}
+
+	newMultiSigGame(player: PublicKey, opponent: PublicKey): Transaction {
+		const admin = multiSigPublicKey([player, opponent]);
+		const tx = new Transaction();
+
+		const game = tx.moveCall({
+			target: `${this.packageId}::owned::new`,
+			arguments: [
+				tx.pure.address(player.toSuiAddress()),
+				tx.pure.address(opponent.toSuiAddress()),
+				tx.pure(bcs.vector(bcs.u8()).serialize(admin.toRawBytes()).toBytes()),
+			],
+		});
+
+		tx.transferObjects([game], admin.toSuiAddress());
+
+		return tx;
+	}
+
+	placeMark(game: Game, row: number, col: number): Transaction {
+		if (game.kind !== 'shared') {
+			throw new Error('Cannot place mark directly on owned game');
+		}
+
+		const tx = new Transaction();
+
+		tx.moveCall({
+			target: `${this.packageId}::shared::place_mark`,
+			arguments: [tx.object(game.id), tx.pure.u8(row), tx.pure.u8(col)],
+		});
+
+		return tx;
+	}
+
+	sendMark(cap: TurnCap, row: number, col: number): Transaction {
```

Comment:
It's a good react pattern to use `mutations` for any "POST" style transactions (user actions)! 

That helps:
1. Keep all mutations in one place
2. Add easier loading states / error handling.

Check this example for some mutations: https://github.com/MystenLabs/sui/blob/d5e1b5fa8625e1f75df511e62453d25b862e94e0/examples/trading/frontend/src/mutations/escrow.ts#L15

Reply:
This particular hook doesn't perform any mutations, it just builds transactions, but it sounds like my `useExecutor` hook is a mutation (it is built on top of `useSignAndExecuteTransaction` which is itself built on `useMutation`, I believe).

Reply:
Made these changes `useExecutor` is moved to the `mutations` directory, and it exposes the various status-related fields. It felt proper to make use of these, so I added some logic to handle pending transactions for "Play" buttons.